KsApi/MockService.swift

```diff
@@ -1175,17 +1184,28 @@
       return SignalProducer(value: messageThread)
     }
 
-    internal func postComment(_: String, toProject _: Project) ->
+    internal func deprecatedPostComment(_: String, toProject _: Project) ->
       SignalProducer<DeprecatedComment, ErrorEnvelope> {
-      if let error = self.postCommentError {
+      if let error = self.deprecatedPostCommentError {
         return SignalProducer(error: error)
-      } else if let comment = self.postCommentResponse {
+      } else if let comment = self.deprecatedPostCommentResponse {
+        return SignalProducer(value: comment)
+      }
+      return .empty
+    }
+
+    func deprecatedPostComment(_: String,
+                               toUpdate _: Update) -> SignalProducer<DeprecatedComment, ErrorEnvelope> {
+      if let error = self.deprecatedPostCommentError {
+        return SignalProducer(error: error)
+      } else if let comment = self.deprecatedPostCommentResponse {
         return SignalProducer(value: comment)
       }
       return .empty
     }
 
-    func postComment(_: String, toUpdate _: Update) -> SignalProducer<DeprecatedComment, ErrorEnvelope> {
+    func postComment(input _: PostCommentInput)
+      -> SignalProducer<GraphMutationPostCommentResponseEnvelope, GraphError> {
       if let error = self.postCommentError {
```

Comment:
I updated this to behave similarly to some other mutations we have. We return the error if it exists; otherwise, we return the successful results value or a template of it.